pyomo/solvers/tests/checks/test_BARON.py

```diff
@@ -0,0 +1,32 @@
+"""Tests the BARON interface."""
+import pyutilib.th as unittest
+from pyomo.environ import (ConcreteModel, Constraint, Objective, Var, log10,
+                           minimize)
+from pyomo.opt import SolverFactory, TerminationCondition
+
+# I don't know how you figure out if BARON is available
+baron_available = True

```

Comment:
This check will not distinguish between those two cases. Although I don't remember precisely, I believe I added this because Baron will error if it finds that your license has expired, rather than just falling back to running in an "unlicensed" state. I believe was the case for a while on Jenkins.